src/k_lsm/k_lsm_inl.h

```diff
@@ -82,6 +82,55 @@ k_lsm<K, V, Rlx>::insert(const K &key,
 
 template <class K, class V, int Rlx>
 bool
+k_lsm<K, V, Rlx>::delete_min(K &key, V &val)
```

Comment:
I agree. There is no reason to duplicate all that code. Will fix that and split out the delete_min changes to a separate commit.